lib/commands/setup.js

```diff
@@ -570,29 +570,69 @@ SetupScreens.prototype.check = function check(callback) {
 					var r = {
 						online: err ? null : online,
 						proxy: config.get('cli.httpProxyServer'),
-						http: false,
-						https: false
+						unreachable: [],
+						javaResults: []
 					};
 
 					if (!r.online) {
 						return next(null, r);
 					}
 
-					// test http and https
-					async.parallel(['http', 'https'].map(function (protocol) {
-						return function (cb) {
-							request({
-								url: protocol + '://api.appcelerator.com/p/v1/release-list',
-								proxy: config.get('cli.httpProxyServer'),
-								rejectUnauthorized: config.get('cli.rejectUnauthorized', true)
-							}, function (error, response, body) {
-								if (!error) {
-									r[protocol] = true;
-								}
+					/*
+						Test network access and proxy permissions via
+						node, cURL, and Java through an async series
+						of tests. Start by attempting to access a set
+						of necessary http and https endpoints.
+					*/
+					async.parallel(JSON.parse(fs.readFileSync(path.resolve(__dirname, 'lib', 'urls_to_check.json'))).map(function (testUrl) {
+							return function (cb) {
+								request({
+									url: testUrl,
+									proxy: config.get('cli.httpProxyServer', ''),
+									rejectUnauthorized: config.get('cli.rejectUnauthorized', true)
+								}, function (error, response, body) {
+									if (error || (response.statusCode && response.statusCode != 200 && response.statusCode != 401)) {
+										// if there's an error, response will be null, treat it as a 404
+										var statCode = (response && response.statusCode) ? response.statusCode : '404'
+										r.unreachable.push(testUrl + ' (HTTP status: ' + statCode + ')');
+									}
+								});
 								cb();
-							});
-						};
-					}), function () {
+							}
+						}),
+					function () {
+						// test access via Java for Studio
+						var javaArgs = ['-jar', path.resolve(__dirname, 'lib/dashboard-login-1.0.0.jar')],
+							outObj;
+						if (config.get('cli.httpProxyServer')) {
+							var proxyParts = config.get('cli.httpProxyServer').split(':');
+							if(proxyParts && proxyParts.length) {
+								if(proxyParts.length === 2) {
+									javaArgs.push('-D' + proxyParts[0] + ".proxyHost=" + proxyParts[1].replace('\/\/', ''));
+								}
+								else if(proxyParts.length > 2) {
+									javaArgs.push('-D' + proxyParts[0] + ".proxyPort=" + proxyParts[2]);
+								}
+							}
+						}
+						appc.subprocess.findExecutable(config.get('executables.java', 'java'), function (err, executable) {
+							if (err) {
+								// skip java tests
+							} else {
+								appc.subprocess.run(executable, javaArgs, function (code, out, err) {
+									if(err) {
+										r.javaResults.push('dashboard.appcelerator.com');
+									} else {
+										outObj = (out) ? JSON.parse(out.substr(out.indexOf("\n") + 1)) : {};
+										if(!outObj.success) {
+											r.javaResults.push('dashboard.appcelerator.com');
```

Comment:
I don't quite understand why are we adding dashboard.appcelerator.com here and at the err condition. Doesn't Java executable return the list of unreachable hosts ?